kratos/geometries/nurbs_curve_geometry.h

```diff
@@ -440,6 +438,34 @@ class NurbsCurveGeometry : public Geometry<typename TContainerPointType::value_t
     ///@name Private Operations
     ///@{
 
+
+    /*
+    * @brief Checks if the knot vector is coinciding with the number of
+    *        control points and the polynomial degree. If the knot vector
+    *        has a multiplicity of p+1 in the beginning, it is reduced to p.
+    */
+    void CheckAndFitKnotVectors()
+    {
+        SizeType number_of_cps = this->size();
+
+        if (mKnots.size() != NurbsUtilities::GetNumberOfKnots(PolynomialDegree, number_of_cps))
```

Comment:
```suggestion
        if (mKnots.size() != NurbsUtilities::GetNumberOfKnots(PolynomialDegree, number_of_cps)) {
```

small but please put the `{` on the same line, check also the rest of the function